src/components/MultiComboBox/useListbox.tsx

```diff
@@ -0,0 +1,316 @@
+        return
+      }
+      const nextIndex = (() => {
+        if (currentIndex === null) {
+          if (delta === 1) {
+            return 0
+          } else {
+            return options.length - 1
+          }
+        }
+        return (currentIndex + delta + options.length) % options.length
+      })()
+      const nextActive = options[nextIndex]
+      if (nextActive && nextActive.disabled) {
+        // skip disabled item
+        moveActiveOptionIndex(nextIndex, delta)
+        return
+      }
+      setActiveOptionIndex(nextIndex)
+    },
+    [options],
+  )
+
+  useEffect(() => {
+    // correct the focus index in dropdown to fit actual row count
+    if (activeOptionIndex === null) {
+      return
+    }
+    const corrected = Math.max(Math.min(activeOptionIndex, options.length - 1), 0)
+    const active = options[corrected]
+    if (!active || active.disabled) {
+      setActiveOptionIndex(null)
+      return
+    }
+    setActiveOptionIndex(corrected)
+  }, [activeOptionIndex, options])
+
+  const handleInputKeyDown = useCallback(
+    (e: React.KeyboardEvent<HTMLInputElement>) => {
+      switch (e.key) {
+        case 'ArrowDown':
+        case 'Down':
+          e.preventDefault()
+          moveActiveOptionIndex(activeOptionIndex, 1)
+          break
+        case 'ArrowUp':
+        case 'Up':
+          e.preventDefault()
+          moveActiveOptionIndex(activeOptionIndex, -1)
+          break
+        case 'Enter': {
+          e.preventDefault()
+          if (activeOptionIndex === null) {
+            return
+          }
+          const activeOption = options[activeOptionIndex]
+          if (activeOption.isAdding) {
+            onAdd && onAdd(activeOption.label)
+            return
+          }
+          if (activeOption && !activeOption.disabled) {
+            const { value, label } = activeOption
+            onSelect({ value, label })
+          }
+          break
+        }
+      }
+    },
+    [activeOptionIndex, moveActiveOptionIndex, onAdd, onSelect, options],
+  )
+
+  const listboxId = useId()
+  const addingButtonId = useId()
+  const optionIdPrefix = useId()
+  const getOptionId = useCallback(
+    (option: Option) => {
+      if (option.isAdding) {
+        return addingButtonId
+      }
+      return `${optionIdPrefix}-${option.label}`
+    },
+    [addingButtonId, optionIdPrefix],
+  )
+  const activeDescendant = (() => {
+    if (activeOptionIndex === null) {
+      return undefined
+    }
+    const activeOption = options[activeOptionIndex]
+    if (activeOption) {
+      return getOptionId(activeOption)
+    }
+    return undefined
+  })()
+
+  const theme = useTheme()
+  const { portalRoot } = usePortal()
+  const listboxRef = useRef<HTMLDivElement>(null)
+
+  const renderListbox = () => {
+    return createPortal(
+      <Container
+        {...dropdownStyle}
+        themes={theme}
+        id={listboxId}
+        className={isExpanded ? undefined : 'hidden'}
```

Comment:
`className` よりもセマンティックな `aria-hidden` を使うのはいかがでしょう？

```suggestion
        aria-hidden={!isExpanded}
```

Reply:
こちら対応し、同じく `className="active"` で選択中の option を表現している部分も `aria-selected` にしたほうが良さそうに思い修正しました！
3b607e2
5210523

Reply:
めちゃ素晴らしい！

WAI-ARIA 1.2 見返すと`role="option"` には `aria-selected` が required になってましたね。 https://www.w3.org/TR/wai-aria-1.2/#option